service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcStructuredAccessException.java

```diff
@@ -0,0 +1,7 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+public class GpcStructuredAccessException extends RuntimeException {
```

Comment:
I had mentioned in another comment throwing a "GpConnectException". Perhaps rename this to "GpConnectException"? 

You'll probably want/need to add the construct passing the root cause - we don't want to throw that away.